src/v/archival/ntp_archiver_service.cc

```diff
@@ -711,6 +722,27 @@ ss::future<> ntp_archiver::upload_until_term_change() {
             break;
         }
 
+        // If a flush was in progress and the archiver has uploaded past the set
+        // _flush_uploads_offset, there should be a number of actions taken to
+        // fully complete the flush operation.
+        // 1. Upload topic manifest (if we are partition 0).
+        // 2. Upload the partition manifest.
+        // 3. Flush manifest to the clean offset.
+        // 4. Reset flush state.
+        // 5. Broadcast using flush cv to alert all waiters.
+        if (uploaded_data_past_flush_offset()) {
+            if (_parent.ntp().tp.partition == 0) {
+                co_await upload_topic_manifest();
+            }
+
+            std::ignore = co_await upload_manifest(
+              upload_loop_flush_complete_ctx_label);
```

Comment:
add a comment about why we are ignoring the return value..

Reply:
Changed this part of the code to no longer ignore the return value of `upload_manifest()`.

We now explicitly check that the return value is `cloud_storage::upload_result::success` before resetting flush state and alerting waiters (i.e completing the flush). Otherwise, we retry the partition manifest upload on the next iteration of the background loop.